### ISSUE TITLE
10.4.0-29E Native Firm

### ISSUE THREAD

**Issue by Shadowtrance** [44670/BootNTR#36]
Sysnand version - 9.0.0-20E
Emunand version - 10.4.0-29E with 10.4 Native Firm 2, 50, 11 (CakeFW latest build)
kversion - 02320b00

result.txt output
```
strHidUser: 003260df
refHidUser: 0030c794
hidObj: 00360da0
{'gxFlushDCahce': '0x12ded0', 'fsOpenFile': '0x163898', 'mountArchive': '0x12c064', 'userFsTryOpen': '0x22bdbc', 'nnInitCardUpd': '0x118d78', 'regArchive': '0x228338', 'fsOpenArchive': '0x21ee10', 'fsWriteFile': '0x163974', 'mountRom': '0x10ed84', 'hidObj': '0x360da0', 'fsReadFile': '0x12c19c', 'getServiceHandle': '0x2359cc', 'aptStartApplet': '0x12ea08', 'fsUserHandle': '0x32efa4'}
if (t == 0xea00001f ) {
ntrConfig->HomeMenuVersion = SYSTEM_VERSION(INSERT_YOUR_VERSION_HERE);
ntrConfig->HomeMenuInjectAddr = 0x12ded0;
ntrConfig->HomeFSReadAddr = 0x12c19c;
ntrConfig->HomeCardUpdateInitAddr = 0x118d78;
ntrConfig->HomeFSUHandleAddr = 0x32efa4;
ntrConfig->HomeAptStartAppletAddr = 0x12ea08;
}
```

**Comment by behdad1999** [comment]
apparently they say that if you download the latest build of ntr it should work, im not really sure.

**Comment by Shadowtrance** [comment]
No, this is the ACTUAL 10.4 Native Firm that was updated when 10.4 came out.
The latest builds support 10.4 emunand/whatever but only 10.2 Native Firm. It's different.
I know this because i released the latest version... :P

**Comment by 44670** [comment]
the 10.4 firm will probably skipped since 10.2 native firm is still workable.

**Comment by Shadowtrance** [comment]
Probably. Don't really see any need for 10.4 native firm yet. Don't think any games use it (i may be wrong but i doubt it at this point).
I was wondering though, how do you go about adding new versions to the kernelversion part of this (bnInitParamsByFirmware)?

**Comment by thedax** [comment]
@Shadowtrance: I'm not 100% sure, but you'd probably have to stick the kernel in IDA or something, and make an entry similar to this once you found the offsets.

https://github.com/44670/BootNTR/blob/master/source/main.c#L298-L307

**Comment by Shadowtrance** [comment]
@thedax Yeah i have no idea what to do in IDA except scratch my head. haha
I'll leave it to someone else, just curious about how to add/fix that part really. Would still like to know of course :)

**Comment by Margen67** [comment]
10.2 may be able to boot the newest home menu, but it requires firmware files on the SD card.
With 10.4 support no firmware files on the SD card would be needed.

**Comment by AuroraWright** [comment]
Bumping this. Since loading FIRM from CTRNAND is now possible (this makes booting faster, among the other things), the latest AuReiNand supports it and I think it's going to eventually spread to all the other CFWs, I think it now makes sense to support 10.4 FIRM.
Still your choice, obviously. If you need the CFW to disable the pseudo ASLR to make NTR work more easily, I'm sure it can be done.

**Comment by TuxSH** [comment]
Yifanlu's loader don't even implement ASLR anyways.

**Comment by fox8091** [comment]
Poke.

**Comment by fox8091** [comment]
Bump.

**Comment by TuxSH** [comment]
Also note that svcBackdoor11 has been reimplemented in Luma3DS, if it's the problem (for 11.0 firm).

**Comment by axelhei** [comment]
dump
